### PR TITLE
Fix: average pool

### PR DESCRIPTION
**Pull request by ivkalgin**
* fixed avg pool operation
* added tests (avg pool)

### CODE REVIEW

File: onnx2torch/node_converters/conv.py [hunk 1]
```diff
@@ -37,25 +38,17 @@ def _(node: OnnxNode, graph: OnnxGraph) -> OperationConverterResult:
     node_attributes = node.attributes
     kernel_size = node_attributes.get('kernel_shape', weights.shape[2:])
     stride = node_attributes.get('strides', 1)
-    padding = node_attributes.get('pads', [0]*4)
     dilation = node_attributes.get('dilations', 1)
     groups = node_attributes.get('group', 1)
 
+    padding = onnx_padding_to_torch_padding(
+        node_attributes.get('pads', [0] * spatial_rank * 2),
+        node_attributes.get('auto_pad', 'NOTSET')
```
Comment (senysenyseny16) [inline onnx2torch/node_converters/conv.py:46]: `,`

File: onnx2torch/node_converters/max_pool.py [hunk 2]
```diff
@@ -27,31 +28,23 @@ def _(node: OnnxNode, graph: OnnxGraph) -> OperationConverterResult:
...
-            raise NotImplementedError(f'Only symmetric padding is implemented ({padding})')
-
-        padding = padding[:half_len]
-    elif auto_pad in ('SAME_UPPER', 'SAME_LOWER', 'VALID'):
-        raise NotImplementedError(f'"{auto_pad}" auto_pad is not implemented')
-    else:
-        raise ValueError(f'Got unexpected auto_pad value "{auto_pad}"')
+    padding = onnx_padding_to_torch_padding(
+        node_attributes.get('pads', [0] * spatial_rank * 2),
+        node_attributes.get('auto_pad', 'NOTSET')
```
Comment (senysenyseny16) [inline onnx2torch/node_converters/max_pool.py:46]: `,`

File: onnx2torch/common.py [hunk 3]
```diff
@@ -62,3 +62,18 @@ def old_style_broadcast(first: torch.Tensor, second: torch.Tensor, axis: int) ->
     second_shape = second_shape + [1]*(rank - len(second_shape))
 
     return second.view(second_shape)
+
+
+def onnx_padding_to_torch_padding(padding: Tuple[int, ...], auto_pad: str) -> Tuple[int, ...]:
+    if auto_pad == 'NOTSET':
+        half_len = len(padding) // 2
+        if tuple(padding[:half_len]) != tuple(padding[half_len:]):
```
Comment (senysenyseny16) [inline onnx2torch/common.py:70]: rm tuple?

File: tests/node_converters/average_pool_max_pool_test.py [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+        ([2, 3, 16, 16, 16], [3, 2, 1], {}),
+        ([2, 3, 16, 16, 16], [2, 2, 2], {'strides': [1, 2, 3]}),
+        ([2, 3, 16, 16, 16], [2, 2, 2], {'ceil_mode': 1}),
+    ),
+)
+def test_max_pool_average_pool(op: str, input_shape: List[int], kernel_shape: List[int], optional_attrs: Dict) -> None:
+    if op == 'AveragePool':
+        optional_attrs['atol_onnx_torch'] = 10**-7
+
+    _test_op('MaxPool', input_shape=input_shape, kernel_shape=kernel_shape, **optional_attrs)
```
Comment (irkjero) [inline tests/node_converters/average_pool_max_pool_test.py:71]:
```python
_test_op(op, input_shape=input_shape, kernel_shape=kernel_shape, **optional_attrs)
```

**Review by irkjero — APPROVED** [review]

**Pull request merged**